src/image/ui/imageballoon.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module image/image/ui/imageballoon
+ */
+
+import ContextualBalloon from '@ckeditor/ckeditor5-ui/src/panel/balloon/contextualballoon';
+import BalloonPanelView from '@ckeditor/ckeditor5-ui/src/panel/balloon/balloonpanelview';
+import { isImageWidget } from '../utils';
+
+/**
+ * A balloon used by various image features to display toolbars, editing forms
+ * etc.
+ *
+ * @extends module:ui/panel/balloon/contextualballoon~ContextualBalloon
+ */
+export default class ImageBalloon extends ContextualBalloon {
```

Comment:
I'm little confused. I don't understand why we need another instance of contextual toolbar. I suppose we don't want to show two toolbars at the same time, so it should be possible to use CT for showing image balloons. If you don't want to duplicate some code for two very similar image balloons you can move it to some helpers.